hollow/src/main/java/com/netflix/hollow/core/index/HollowPrimaryKeyIndex.java

```diff
@@ -276,13 +277,18 @@ public int getMatchingOrdinal(Object key1, Object key2, Object key3) {
      */
     public int getMatchingOrdinal(Object... keys) {
         PrimaryKeyIndexHashTable hashTable = hashTableVolatile;
-        if(fieldPathIndexes.length != keys.length || hashTable.bitsPerElement == 0)
+
+        // for special case of keys containing one null element
+        if (keys == null || (keys != null && fieldPathIndexes.length != keys.length) || hashTable.bitsPerElement == 0)
```

Comment:
This will throw NPE for cases like getMatchingOrdinal(null). 

getMatchingOrdinal((Object) null), getMatchingOrdinal(null, null), getMatchingOrdinal(new Object[]{null, null}) and so on will pass this check.

Reply:
The `keys != null` check is redundant, it can be dropped

Reply:
Done.